### DISCUSSION TITLE
Script Ideas / Feature Requests

### DISCUSSION THREAD

**Discussion by topper-2k** [PinballY/PinballY-Addons-and-Examples#4] [Ideas]
Thanks so much to the very talented folks who have dedicated their time and expertise to write the scripts in this repository! 

This discussion thread is intended as a place where folks (programmers and non-programmers) can float ideas, features, etc. that might make PinballY even better and get feedback from others about those ideas.

This isn't a place to get support with using PBY features. If you need help, try the Facebook group or one of the forums (VPForums or VPinball for example). 

Cheers!

**Comment by topper-2k** [comment]
Ok, this seems trivial, but one thing I find surprisingly inconvenient when using PBY is that the "Game Setup" menu option moves after a table's details have been entered the first time. It doesn't feel like it belongs under "Operator Menu." 

A script that added "Game Setup" as an option to the main menu (right after "Play" or Play (version) if using the 2 games 1 wheel script) would make table management much easier and eliminate several button presses from the task of capturing media, assigning categories, changing game descriptions, etc.

**Reply by worksasdesigned** [reply]
Check this:
https://github.com/PinballY/PinballY-Addons-and-Examples/tree/main/command_example

Its a simple pattern which allows to add any of the predifined commands to the main menu.

**Comment by topper-2k** [comment]
One more while I'm at it ;) A script that would let us edit the machine types would be awesome. For example, there are A LOT of tables that are born digital (most of FX3, most of Scotty Wic's collection, etc.) It feels weird to classify Star Wars Rogue One or Pizza Time as "Solid State." I'd love the option of editing that list to my liking, so I could add something like "Digital" or "Original" to the list of options.

**Reply by worksasdesigned** [reply]
i have no idea where those predefined types are stored. With java script i don't think that we can influence them.

**Reply by mjrgh** [reply]
> With java script i don't think that we can influence them.

You're right - that's a hard-wired internal list right now.  It comes from the HyperPin classifications, which come from the IPDB classifications, so it's kind of hard-wired at a "conceptual" level, too.

I can see how it might seem a little strange to apply the IPDB classes to a VP original that was never built physically; it's obviously not a *literal* SS or EM game. On the other hand, nearly all of the pure-virtual games I've seen were still meant to *simulate* a particular class of real pinball machines.  SWRO is clearly meant to be a DMD-era solid-sate game, for example, and PT is meant to be an alphanumeric era SS.  To some extent, if someone came up with something that *didn't* fit one of the IPDB classes at least approximately, I'm not sure it would even be a pinball simulation!

And as a practical matter, I'm not sure a new class for "pure virtual" would actually be good, because you'd lose that information that you get from ME/EM/SS about the score display type.  (And the IPDB categories aren't great for that, since they're not fine-grained enough to tell you whether you're talking about DMD, 16-segment, 7-segment, reels, score lamps, or something else.)  What would be more useful, I think, would be to treat pure-virtualness as an orthogonal bit of metadata - a category tag would be one easy way to handle it with existing features.
